x/incentives/keeper/distribute_test.go

```diff
@@ -1376,3 +1450,174 @@ func (s *KeeperTestSuite) TestCreateGroupGaugeAndDistribute() {
 	}
 
 }
+
+// deepCopyGroupGauge creates a deep copy of the passed in group gauge.
+func deepCopyGroupGauge(src types.GroupGauge) types.GroupGauge {
+	gaugeRecords := make([]types.InternalGaugeRecord, len(src.InternalGaugeInfo.GaugeRecords))
+	for i, record := range src.InternalGaugeInfo.GaugeRecords {
+		gaugeRecords[i] = types.InternalGaugeRecord{
+			GaugeId:          record.GaugeId,
+			CurrentWeight:    record.CurrentWeight,
+			CumulativeWeight: record.CumulativeWeight,
+		}
+	}
+
+	return types.GroupGauge{
+		GroupGaugeId: src.GroupGaugeId,
+		InternalGaugeInfo: types.InternalGaugeInfo{
+			TotalWeight:  src.InternalGaugeInfo.TotalWeight,
+			GaugeRecords: gaugeRecords,
+		},
+		SplittingPolicy: src.SplittingPolicy,
+	}
+}
+
+// withUpdatedVolumes takes in a group gauge and a list of updated cumulative volumes (ordered) and updates the contents of the gauge to
+// reflect these new volumes.
+// It is only intended to be used to set expected values for test cases.
+func (s *KeeperTestSuite) withUpdatedVolumes(groupGauge types.GroupGauge, updatedCumulativeVolumes []osmomath.Int) types.GroupGauge {
+	// Ensure there aren't more volumes to update than records in group gauge
+	s.Require().True(len(updatedCumulativeVolumes) <= len(groupGauge.InternalGaugeInfo.GaugeRecords))
+
+	// We make a deep copy of the group gauge to ensure we don't modify the original input/defaults
+	updatedGroupGauge := deepCopyGroupGauge(groupGauge)
+
+	newTotalWeight := osmomath.ZeroInt()
+	for i, updatedVolume := range updatedCumulativeVolumes {
+		currentRecord := groupGauge.InternalGaugeInfo.GaugeRecords[i]
+		updatedRecord := types.InternalGaugeRecord{
+			GaugeId:          currentRecord.GaugeId,
+			CurrentWeight:    updatedVolume.Sub(currentRecord.CumulativeWeight),
+			CumulativeWeight: updatedVolume,
+		}
+		updatedGroupGauge.InternalGaugeInfo.GaugeRecords[i] = updatedRecord
+		newTotalWeight = newTotalWeight.Add(updatedRecord.CurrentWeight)
+	}
+	updatedGroupGauge.InternalGaugeInfo.TotalWeight = newTotalWeight
+
+	return updatedGroupGauge
+}
+
+// withSplittingPolicy returns a deep copy of the passed in group gauge with the splitting policy set to the passed in value.
+func withSplittingPolicy(groupGauge types.GroupGauge, splittingPolicy types.SplittingPolicy) types.GroupGauge {
+	// We make a deep copy of the group gauge to ensure we don't modify the original input/defaults
+	updatedGroupGauge := deepCopyGroupGauge(groupGauge)
+	updatedGroupGauge.SplittingPolicy = splittingPolicy
```

Comment:
nit: I don't think deep copy is necessary here since `SplittingPolicy` is a not a field allocated in the dynamic memory

Reply:
The deep copy here serves a slightly different purpose. Since the system under test mutates the input, we need to generate a new deep copy for every case to avoid mutating default values. Technically we can do a shallow copy here and then do a deep copy of that in the test field, but this felt unnecessarily verbose since pretty much all uses of updated group gauges will require a deep copy.

If I'm missing something here and there's a more elegant way to do this please lmk (even if it's a nit)!